app/validate.py

```diff
@@ -0,0 +1,138 @@
+    if re.match(UUID_PATTERN, uid):
+        webhook.logger.info("Valid UID Found, continuing")
+    else:
+        message = "Invalid UID. Aborting validation"
+        webhook.logger.error("K8s UID failed security checks. Request rejected!")
+        return admission_response(allowed=False, uid=uid, message=message)
+
+    # check we're not in a system namespace
+    if namespace in webhook.extensions["whorf"].ignores_namespaces:
+        message = "Namespace in ignore list. Ignoring validation"
+        webhook.logger.error("Namespace in ignore list. Ignoring validation!")
+        return admission_response(allowed=True, uid=uid, message=message)
+
+    return None
+
+
+def process_passed_checks(ckv_whorf: CheckovWhorf, uid: str, obj_kind_name: str) -> Response:
+    """Invoked when no Kubernetes related issues were found"""
+
+    message = []
+
+    k8s_message = generate_k8s_output(reports=ckv_whorf.scan_reports)
+    message.extend(k8s_message)
+
+    sca_message = generate_sca_output(reports=ckv_whorf.scan_reports)
+    message.extend(sca_message)
+
+    webhook.logger.info(f"Object {obj_kind_name} passed security checks. Allowing the request.")
+    return admission_response(allowed=True, uid=uid, message="\n".join(message))
+
+
+def process_failed_checks(ckv_whorf: CheckovWhorf, uid: str, obj_kind_name: str) -> Response:
+    """Invoked when Kubernetes related issues were found"""
+
+    message = []
+
+    if ckv_whorf.config.hard_fail_on:
+        hard_fails = {}
+        try:
+            for report in ckv_whorf.scan_reports:
+                for check in report.failed_checks:
+                    if check.check_id in ckv_whorf.config.hard_fail_on:
+                        hard_fails[check.check_id] = f"\n  Description: {check.check_name}"
+                        if check.guideline:
+                            hard_fails[check.check_id] += f"\n  Guidance: {check.guideline}"
+                    elif check.bc_check_id in ckv_whorf.config.hard_fail_on:
+                        hard_fails[check.check_id] = f"\n  Description: {check.check_name}"
+                        if check.guideline:
+                            hard_fails[check.check_id] += f"\n  Guidance: {check.guideline}"
+        finally:
+            webhook.logger.error("hard fail error")
+
+        if hard_fails:
+            message.append(f"Checkov found {len(hard_fails)} issues in violation of admission policy.")
+
+            for ckv in hard_fails:
+                message.append(f"{ckv}:{hard_fails[ckv]}")
+
+    k8s_message = generate_k8s_output(reports=ckv_whorf.scan_reports)
+    message.extend(k8s_message)
+
+    sca_message = generate_sca_output(reports=ckv_whorf.scan_reports)
+    message.extend(sca_message)
+
+    webhook.logger.error(f"Object {obj_kind_name} failed security checks. Request rejected!")
+    return admission_response(allowed=False, uid=uid, message="\n".join(message))
```

Comment:
@gruebel If soft-fail is configured, shouldn't the allowed be `True`?

Reply:
separate issue created #8